### PR TITLE
Rpi 4.4.y: HDMI hotplug detection, overlay cleanup

### PR DESCRIPTION
**Pull request by anholt**
I finally got a rework of the overlay to work (mostly just irritating debug from overlays not validating well and silently failing to apply at runtime), so I could get HDMI hotplug detection fixed.  The commits on this branch should apply to 4.5.y as well.

### CODE REVIEW

**Comment by popcornmix** [comment]
@pelwell are you happy with DT changes?

File: arch/arm/boot/dts/bcm2708_common.dtsi [hunk 1]
```diff
@@ -261,6 +282,24 @@
 			status = "disabled";
 		};
 
+		pixelvalve2: pixelvalve@7e807000 {
+			compatible = "brcm,bcm2835-pixelvalve2";
+			reg = <0x7e807000 0x100>;
```
Comment (pelwell) [inline arch/arm/boot/dts/bcm2708_common.dtsi]: I'd like a `status = "disabled";` here.

**Comment by pelwell** [comment]
Yes, apart from the fact that the pixelvalve2 node is enabled by default. The overlay already contains the necessary status="okay" to enable it, so I just want a single line added.

**Comment by anholt** [comment]
Re-pushed the branch with pixelvalve fixed up.

**Comment by pelwell** [comment]
Looks good to me.

**Pull request merged**